app/main/posts/modify/post-media.directive.js

```diff
@@ -17,84 +19,44 @@ function (
-            $scope.fileContainer = {file: null};
+            $scope.showAdd = function () {
+                return (!$scope.media.id && !$scope.media.changed);
+            };
 
-            // Initialize media object
-            $scope.media = {};
+            $scope.showReplace = function () {
+                return $scope.media.dataURI || $scope.media.id;
+            };
+
+            $scope.showDelete = function () {
+                return $scope.media.id;
+            };
 
             if ($scope.mediaId) {
                 MediaEndpoint.get({id: $scope.mediaId}).$promise.then(function (media) {
                     $scope.media = media;
+                    // Set initial media state
+                    $scope.media.changed = false;
                 });
+            } else {
+                // Initialize media object
+                $scope.media = {file: null, caption: null, dataURI: null, changed: false};
             }
 
-            // Track file changes
-            $scope.canUpload = false;
-
-            $scope.onChange = function () {
-                $scope.$apply(function () {
-                    if ($scope.fileContainer.file) {
-                        $scope.canUpload = true;
-                    } else {
-                        $scope.canUpload = false;
-                    }
-                });
-            };
-
-            $scope.uploadFile = function () {
-                //@todo Allow editing of caption for existing image
-                if (!$scope.fileContainer.file) {
-                    return;
-                }
-
-                // Delete current file
-                var promise = deleteMedia($scope.mediaId);
-
-                // ...then upload new file
-                promise.then(function () {
-                    var formData = new FormData();
-
-                    formData.append('file', $scope.fileContainer.file);
-
-                    if ($scope.media.caption) {
-                        formData.append('caption', $scope.media.caption);
-                    }
-
-                    $http.post(
-                        Util.apiUrl('/media'),
-                        formData,
-                        {
-                            headers: {
-                                'Content-Type': undefined
-                            }
-                        }
-                    ).then(function (response) {
-                        $scope.mediaId = response.data.id;
-
-                        // We found a file so change the state of parent form
-                        formCtrl[$scope.name].$setDirty();
-                    }, function (error) {
-                        Notify.apiErrors(error);
+            $scope.deleteMedia = function (mediaId) {
+                // Mark for deletion
+                Notify.confirmDelete('notify.notification.delete_confirm').then(function () {
+                    MediaEditService.deleteMedia(mediaId).then(function () {
```

Comment:
hrm ok you make a good point. If we were to delay delete we'd probably have to include an 'undo' function and then scope creeps larger.

I don't like that this means a 'cancel' on post edit doesn't actually cancel everything because of this.. but I'm ok with leaving it for now.